src/ngraph/runtime/cpu/pass/rnn_fusion.cpp

```diff
@@ -0,0 +1,566 @@
+        {
+            // lstm's will be the input to GOE's
+            lstm_nodes.push_back(lstm_goes[i]->get_arguments()[0]);
+        }
+
+        if (sequence_len != lstm_nodes.size())
+        {
+            throw ngraph_error(" Number of lstm nodes in RNN layer is not equal to time slices");
+        }
+
+        if (lstm_nodes.size() != lstm_goes.size() &&
+            lstm_goes.size() != ht_slice_per_timestep.size())
+        {
+            throw ngraph_error(
+                "Number of slices of rnn output ht is not equal to the time slices in RNN layer");
+        }
+
+        // collect all the consumers of LSTM goe's (ht)
+        std::set<std::shared_ptr<ngraph::Node>> lstm_goe0_user;
+        std::unordered_map<std::shared_ptr<Node>, std::shared_ptr<Node>> map_goe_to_lstm_slices;
+        std::shared_ptr<Node> goe_0;
+        for (size_t index = 0; index < lstm_nodes.size(); index++)
+        {
+            // now get the GOE0 which is the first output of lstm (ht)
+            for (auto& goes : lstm_nodes[index]->get_outputs().at(0).get_inputs())
+            {
+                auto goe_node = std::dynamic_pointer_cast<op::GetOutputElement>(goes->get_node());
+                // first output node of lstm
+                if (goe_node->get_n() == 0)
+                {
+                    goe_0 = goes->get_node();
+                }
+            }
+
+            for (auto goe0_user : goe_0->get_users())
+            {
+                if (std::find(lstm_nodes.begin(), lstm_nodes.end(), goe0_user) ==
+                        lstm_nodes.end() &&
+                    !is_unreachable(goe0_user))
+                {
+                    lstm_goe0_user.insert(goe0_user);
+                    map_goe_to_lstm_slices[goe_0] = ht_slice_per_timestep[index];
+                    NGRAPH_DEBUG << "ht_slice: " << ht_slice_per_timestep[index]->get_name()
+                                 << " goe0_user " << goe0_user->get_name() << " ";
+                }
+            }
+        }
+
+        //now go through the lstm consumers and replace them with the slice
+        std::vector<std::shared_ptr<Node>> new_args;
+        for (auto& node : lstm_goe0_user)
+        {
+            for (auto& node_args : node->get_arguments())
+            {
+                if (map_goe_to_lstm_slices.find(node_args) != map_goe_to_lstm_slices.end())
+                {
+                    NGRAPH_DEBUG << " args_shape " << join(node_args->get_shape())
+                                 << "   name: " << node_args->get_name();
+                    new_args.push_back(map_goe_to_lstm_slices[node_args]);
+                }
+                else
+                {
+                    new_args.push_back(node_args);
+                }
+            }
+            NGRAPH_DEBUG << "node bring replaced " << node->get_name();
+            auto new_node = node->copy_with_new_args(new_args);
+            if (!std::dynamic_pointer_cast<op::Result>(node))
```

Comment:
nitpick : `node->is_output`

Reply:
Add support for the case if the o/p of GOE is Result Op